src/sentry/search/events/filter.py

```diff
@@ -385,52 +385,48 @@ def _semver_filter_converter(
-        .order_by(*order_by)[:MAX_SEARCH_RELEASES]
-    )
-    versions = list(qs)
-    final_operator = "IN"
-    if len(versions) == MAX_SEARCH_RELEASES:
-        # We want to limit how many versions we pass through to Snuba. If we've hit
-        # the limit, make an extra query and see whether the inverse has fewer ids.
-        # If so, we can do a NOT IN query with these ids instead. Otherwise, we just
-        # do our best.
-        operator = OPERATOR_NEGATION_MAP[operator]
-        # Note that the `order_by` here is important for index usage. Postgres seems
-        # to seq scan with this query if the `order_by` isn't included, so we
-        # include it even though we don't really care about order for this query
-        qs_flipped = (
-            Release.objects.filter_by_semver(organization_id, parse_semver(version, operator))
-            .order_by(*map(_flip_field_sort, order_by))
-            .values_list("version", flat=True)[:MAX_SEARCH_RELEASES]
+    def get_versions(v: str, op: str) -> tuple[str, list[str]]:
+        order_by = Release.SEMVER_COLS
+        if op.startswith("<"):
+            order_by = list(map(_flip_field_sort, order_by))
+        qs = (
+            Release.objects.filter_by_semver(
+                organization_id,
+                parse_semver(v, op),
+                project_ids=project_ids,
+            )
+            .values_list("version", flat=True)
+            .order_by(*order_by)[:MAX_SEARCH_RELEASES]
         )
-
-        exclude_versions = list(qs_flipped)
-        if exclude_versions and len(exclude_versions) < len(versions):
-            # Do a negative search instead
-            final_operator = "NOT IN"
-            versions = exclude_versions
-
-    if not versions:
-        # XXX: Just return a filter that will return no results if we have no versions
-        versions = [SEMVER_EMPTY_RELEASE]
-
-    return ["release", final_operator, versions]
+        versions = list(qs)
+        final_operator = "IN"
+        if len(versions) == MAX_SEARCH_RELEASES:
+            op = OPERATOR_NEGATION_MAP[op]
+            qs_flipped = (
+                Release.objects.filter_by_semver(organization_id, parse_semver(v, op))
+                .order_by(*map(_flip_field_sort, order_by))
+                .values_list("version", flat=True)[:MAX_SEARCH_RELEASES]
+            )
+            exclude_versions = list(qs_flipped)
+            if exclude_versions and len(exclude_versions) < len(versions):
+                final_operator = "NOT IN"
+                versions = exclude_versions
+        if not versions:
+            versions = [SEMVER_EMPTY_RELEASE]
+        return final_operator, versions
+
+    if isinstance(version, str):
+        final_operator, versions = get_versions(version, operator)
+        return ["release", final_operator, versions]
+    else:
+        semver_filters = []
+        for v in version:
+            _, versions = get_versions(v, operator)
```

Comment:
I'm not really sure `get_versions` is necessary for `IN`. Since in will just be doing exact matches, all the sorting junk that we do isn't really relevant.